modules/mgclient.mjs

```diff
@@ -0,0 +1,1829 @@
+        let result =
+            new MgLocalDateTime(wrappedFun(this.#cPtr));
+        return errorOrPushToResourceManager(result);
+    }
+
+    /**
+     * Destructs the memory allocated by the C runtime.
+     * This function is used by the ResourceManager to destruct at the end of execution.
+     */
+    destroy() {
+        let wrappedFun =
+            instance.cwrap('mg_local_date_time_destroy', 'void', ['i32']);
+        wrappedFun(this.#cPtr);
+    }
+
+    /**
+     * This function shall not be used by this module's consumers.
+     *
+     * Internal function to pass the MgValue to the underlyine wasm mgclient.
+     * @return {int}. Memory address of the underlyine object.
+     */
+    transferToWasm() {
+        return this.#cPtr;
+    }
+};
+
+export class MgDuration {
+    #cPtr;
+    constructor(cPtr) {
+        this.#cPtr = cPtr;
+    }
+
+    static make(months, days, seconds, nanoseconds) {
+        let wrappedFun = instance.cwrap('mg_duration_make', 'i32', ['i64', 'i64', 'i64', 'i64']);
+        let result = new MgDuration(instance._mg_duration_make(BigInt(months), BigInt(days), BigInt(seconds), BigInt(nanoseconds)));
+        return errorOrPushToResourceManager(result);
+    }
+
+    months() {
+        let wrappedFun = instance.cwrap('mg_duration_months', 'number', ['i32']);
+        return wrappedFun(this.#cPtr);
+    }
+
+    days() {
+        let wrappedFun = instance.cwrap('mg_duration_days', 'number', ['i32']);
+        return wrappedFun(this.#cPtr);
+    }
+
+    seconds() {
+        let wrappedFun = instance.cwrap('mg_duration_seconds', 'number', ['i32']);
+        return wrappedFun(this.#cPtr);
+    }
+
+    nanoseconds() {
+        let wrappedFun =
+            instance.cwrap('mg_duration_nanoseconds', 'i64', ['i32']);
+        return wrappedFun(this.#cPtr);
+    }
+
+    /**
+     * Returns a copy of the MgDuration (allocates memory on the C runtime)
+     * @return {MgDuration, null}. null is returned if the allocation fails.
+     */
+    copy() {
+        let wrappedFun = instance.cwrap('mg_duration_copy', 'i32', ['i32']);
+        let result = new MgDuration(wrappedFun(this.#cPtr));
+        return errorOrPushToResourceManager(result);
+    }
+
+    /**
+     * Destructs the memory allocated by the C runtime.
+     * This function is used by the ResourceManager to destruct at the end of execution.
+     */
+    destroy() {
+        let wrappedFun = instance.cwrap('mg_duration_destroy', 'void', ['i32']);
+        wrappedFun(this.#cPtr);
+    }
+
+    /**
+     * This function shall not be used by this module's consumers.
+     *
+     * Internal function to pass the MgValue to the underlyine wasm mgclient.
+     * @return {int}. Memory address of the underlyine object.
+     */
+    transferToWasm() {
+        return this.#cPtr;
+    }
+};
+
+// These are not supported yet by mgclient or memgraph but the interfaces are added for completeness
```

Comment:
I would remove all this commented code because it's not used/tested. If there is a way to test without `mgclient`|`memgraph`, I would "disable" the feature on any meaningful level.

Completeness is fine, but I've never seen a bunch of commented code that makes any sense...